Bottom: 918cb6773974f12381812bf6af5f3312e944eabd
Top:    1410829865ebbbc39ab5a0f7fb9020caaecdcd54
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-12-08 10:36:33 +0800

dev: reproducer

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/net/core/dev.c b/net/core/dev.c
index 6bea893..82b5478 100644
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -7674,6 +7674,9 @@ int register_netdevice(struct net_device *dev)
 	else if (__dev_get_by_index(net, dev->ifindex))
 		goto err_uninit;
 
+	if (!strcmp(dev->name, "tap0"))
+		goto err_uninit;
+
 	/* Transfer changeable features to wanted_features and enable
 	 * software offloads (GSO and GRO).
 	 */
